simulator/src/main/java/com/hedera/block/simulator/grpc/PublishStreamGrpcClientImpl.java

```diff
@@ -138,10 +148,45 @@ public boolean streamBlock(Block block) {
                 break;
             }
         }
-
+        publishedBlocks++;
         return streamEnabled.get();
     }
 
+    /**
+     * Sends a onCompleted message to the server and waits for a short period of
+     * time to ensure the message is sent.
+     *
+     * @throws InterruptedException if the thread is interrupted
+     */
+    @Override
+    public void completeStreaming() throws InterruptedException {
+        requestStreamObserver.onCompleted();
+        Thread.sleep(100);
```

Comment:
Do we really need to sleep here? Is the `onCompleted()` method not blocking or am I mistaken?

Reply:
I see, it would be good to investigate this in the future, do you agree to make an issue for the future and tagging it with our `@todo(issueNumber)` convention so we do not forget?